src/network/networkd-dhcp4.c

```diff
@@ -71,8 +71,16 @@ static int link_set_dhcp_routes(Link *link) {
         int r, n, i;
 
         assert(link);
-        assert(link->dhcp_lease);
-        assert(link->network);
+
+        if (link->dhcp_lease == NULL) {
+                log_link_warning(link, "dhcp lease was null. Did the link go down?");
+                return 0;
+        }
+
+        if (link->network == NULL) {
+                log_link_warning(link, "network was null. Did the link go down?");
+                return 0;
+        }
```

Comment:
fix looks conceptually OK, but two nitpicks:

1. please just exit quietly, this is a common, non-exceptional case. just replace the warning log lines with a brief comment: `/* The link went down while we configured the IP addresses? */` or so...

2. we usually check pointers for non-NULL relying on C's degrade-to-bool feature, instead of explicit NULL comparisons, also see CODING_STYLE.

I.e. this should simply read:

```c
if (!link->dhcp_lease) /* The link went down while we configured the IP addresses? */
        return 0;
```

and similar for the network case...